File: android/src/main/java/com/expoalarmmodule/receivers/DismissReceiver.java
package com.expoalarmmodule.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.expoalarmmodule.Manager;

public class DismissReceiver extends BroadcastReceiver {

    private static final String TAG = "AlarmDismissReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        // Retrieve the alarm UID from the intent
        String alarmUid = intent.getStringExtra("ALARM_UID");

        // Stop the active alarm by invoking Manager.stop()
        Manager.stop(context);

        // Optionally log more information about the stop action
        Log.d(TAG, "Alarm " + alarmUid + " has been stopped.");
    }
}

File: android/src/main/java/expo/modules.alarmemodule/receivers/DismissReceiver.kt
package expo.modules.alarmemodule.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class DismissReceiver : BroadcastReceiver() {
    @Override
    fun onReceive(context: Context?, intent: Intent) {
        // TODO: snooze alarm and send event to js
        Log.d(TAG, "dismissed alarm notification for: " + intent.getStringExtra("ALARM_UID"))
    }

    companion object {
        private const val TAG = "AlarmDismissReceiver"
    }
}